packages/zoe/src/instanceRecordStorage.js

```diff
@@ -1,142 +1,103 @@
-import { provide } from '@agoric/vat-data';
+import { provide, vivifyFarClass, M } from '@agoric/vat-data';
 import { assertKeywordName } from './cleanProposal.js';
 
 const { ownKeys } = Reflect;
 
-const { details: X, quote: q, Fail } = assert;
+const { quote: q, Fail } = assert;
 
 /**
  * The InstanceRecord stores the installation, customTerms, issuers,
  * and brands for a particular Zoe contract instance. The installation
  * and customTerms are never changed, but new issuers (and their
  * matching brands) may be added by the contract code. Thus, an
- * InstanceRecord may be outdated at any particular point. This file
- * manages the creation and updating of an InstanceRecord and returns
+ * InstanceRecord may be outdated at any particular point. This class
+ * manages the creation and updating of an InstanceRecord and provides
  * functions for getting the latest data.
  */
 
-/**
- * @callback GetInstanceRecord
- * @returns {InstanceRecord}
- */
-
 /**
  * @param {import('@agoric/vat-data').Baggage} baggage
- * @returns {InstanceRecordManager}
+ * @returns {(ir: InstanceRecord) => InstanceState}
  */
 export const makeInstanceRecordStorage = baggage => {
   provide(baggage, 'instanceRecord', () => undefined);
 
-  const assertInstantiated = () =>
+  const assertInstantiated = instanceRecord =>
     assert(
-      baggage.get('instanceRecord') !== 'undefined',
+      instanceRecord !== 'undefined',
       'instanceRecord has not been instantiated',
     );
 
-  const addIssuerToInstanceRecord = (keyword, issuerRecord) => {
-    !(keyword in issuerRecord) || Fail`conflicting definition of ${q(keyword)}`;
-
-    assertInstantiated();
-    const instanceRecord = baggage.get('instanceRecord');
-    const nextInstanceRecord = harden({
-      ...instanceRecord,
-      terms: {
-        ...instanceRecord.terms,
-        issuers: {
-          ...instanceRecord.terms.issuers,
-          [keyword]: issuerRecord.issuer,
-        },
-        brands: {
-          ...instanceRecord.terms.brands,
-          [keyword]: issuerRecord.brand,
-        },
-      },
-    });
-    baggage.set('instanceRecord', nextInstanceRecord);
-  };
-
-  /** @type {GetInstanceRecord} */
-  const getInstanceRecord = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord');
-  };
-  const getTerms = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').terms;
-  };
-  const getIssuers = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').terms.issuers;
-  };
-  const getBrands = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').terms.brands;
-  };
-  /** @type {InstanceRecordManagerGetInstallationForInstance} */
-  const getInstallationForInstance = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').installation;
-  };
+  const InstanceRecordInterface = M.interface(
```

Comment:
I think the convention is to abbreviate "Interface".
```suggestion
  const InstanceRecordI = M.interface(
```

Reply:
done. Thanks.